github/checks.go

```diff
@@ -0,0 +1,133 @@
+	AnnotationsCount *int               `json:"annotations_count,omitempty"`
+	AnnotationsURL   *string            `json:"annotations_url,omitempty"`
+	Annotations      []*CheckAnnotation `json:"annotations,omitempty"`
+	Images           []*CheckImage      `json:"images,omitempty"`
+}
+
+// CheckAnnotation represents an annotation object for a CheckRun output.
+type CheckAnnotation struct {
+	FileName     *string `json:"filename,omitempty"`
+	BlobHRef     *string `json:"blob_href,omitempty"`
+	StartLine    *int    `json:"start_line,omitempty"`
+	EndLine      *int    `json:"end_line,omitempty"`
+	WarningLevel *string `json:"warning_level,omitempty"`
+	Message      *string `json:"message,omitempty"`
+	Title        *string `json:"title,omitempty"`
+	RawDetails   *string `json:"raw_details,omitempty"`
+}
+
+// CheckImage represents an image object for a CheckRun output.
+type CheckImage struct {
+	Alt      *string `json:"alt,omitempty"`
+	ImageURL *string `json:"image_url,omitempty"`
+	Caption  *string `json:"caption,omitempty"`
+}
+
+// CheckSuite represents a suite of check runs.
+type CheckSuite struct {
+	ID *int64 `json:"id,omitempty"`
+}
+
+func (c CheckRun) String() string {
+	return Stringify(c)
+}
+
+// GetCheckRun gets a check-run for a repository.
+//
+// GitHub API docs: https://developer.github.com/v3/checks/runs/#get-a-single-check-run
+func (s *ChecksService) GetCheckRun(ctx context.Context, owner string, repo string, checkRunID int64) (*CheckRun, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/check-runs/%v", owner, repo, checkRunID)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	acceptHeaders := []string{mediaTypeCheckRunsPreview}
+	req.Header.Set("Accept", strings.Join(acceptHeaders, ", "))
+
+	checkRun := new(CheckRun)
+	resp, err := s.client.Do(ctx, req, checkRun)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return checkRun, resp, nil
+}
+
+// CreateCheckRunOptions sets up parameters need to create a CheckRun.
+type CreateCheckRunOptions struct {
+	Name        *string         `json:"name,omitempty"`
+	HeadBranch  *string         `json:"head_branch,omitempty"`
+	HeadSHA     *string         `json:"head_sha,omitempty"`
+	DetailsURL  *string         `json:"details_url,omitempty"`
+	ExternalID  *int64          `json:"external_id,omitempty"`
+	Status      *string         `json:"status,omitempty"`
+	Conclusion  *string         `json:"conclusion,omitempty"`
+	StartedAt   *Timestamp      `json:"started_at,omitempty"`
+	CompletedAt *Timestamp      `json:"completed_at,omitempty"`
+	Output      *CheckRunOutput `json:"output,omitempty"`
+}
+
+// CreateCheckRun Creates a check run for repository.
+//
+// GitHub API docs: https://developer.github.com/v3/checks/runs/#create-a-check-run
+func (s *ChecksService) CreateCheckRun(ctx context.Context, owner string, repo string, opt *CreateCheckRunOptions) (*CheckRun, *Response, error) {
```

Comment:
According to https://developer.github.com/v3/checks/runs/#create-a-check-run, some of the parameters are required. They're not all optional. So, let's use a value `opt CreateCheckRunOptions ` rather than pointer `opt * CreateCheckRunOptions` here, to avoid having to deal with the bad input `nil`.